### PR TITLE
Make the configuration a bit more trivial

### LINKED ISSUE #1398: 12 factor - use env variables to configure hostname
Modern web app - [12 factor](https://12factor.net/) web app understand env variables, it makes it a lot easier to run them.

It would ease the configuration of this part:
https://github.com/jitsi/jitsi-meet/blob/master/config.js#L6-L18

It would be really nice if jitsi meet would understand env variables to configure these.

### PR DESCRIPTION
**Pull request by pierreozoux**
Closes #1398 
Thanks for your time!

### CODE REVIEW

**Comment by saghul** [comment]
Hey Pierre, thanks! Have you signed our CLA?

**Comment by pierreozoux** [comment]
Yes I did :) But I didn't know how to say it to the bot..

**Comment by saghul** [comment]
@bgrozev can you please check if the CLA signature was processed? (I don't know how to check this myself)

**Comment by saghul** [comment]
At any rate, the change LGTM, thanks for your work! 👍

**Comment by damencho** [comment]
I'm a little bit in doubt here. As this is configured during debian package installation. We need to discuss this with the other devs. And this file is used by the ssi engines inside jetty, nginx and apache. 
If we decide we want this change, we need to simply add the file to the debian package and not use it as template and add it to the /usr/share/doc folder and remove the logic from the debian scripts replacing strings in this file.
I agree this simplifies things a lot, but it needs to be tested with jetty, nginx and apache before stepping forward and we need to modify the debian packaging.

**Comment by saghul** [comment]
@damencho Right! I had forgotten we use search and replace! Thanks for pointing that out! 👍

**Comment by bgrozev** [comment]
> @bgrozev can you please check if the CLA signature was processed?

Yeah, we have the CLA

**Comment by saghul** [comment]
@pierreozoux Can you please adjust this too? https://github.com/jitsi/jitsi-meet/blob/master/debian/jitsi-meet-web-config.postinst#L98 Otherwise our Debian packages would break.

**Comment by pierreozoux** [comment]
done

**Comment by pierreozoux** [comment]
Also edited the manual install to reflect this change.

File: debian/jitsi-meet-web-config.postinst [hunk 1]
```diff
@@ -95,7 +95,6 @@ case "$1" in
         JITSI_MEET_CONFIG="/etc/jitsi/meet/$JVB_HOSTNAME-config.js"
         if [ ! -f $JITSI_MEET_CONFIG ] ; then
             cp /usr/share/doc/jitsi-meet-web-config/config.js $JITSI_MEET_CONFIG
-            sed -i "s/jitsi-meet.example.com/$JVB_HOSTNAME/g" $JITSI_MEET_CONFIG
```
Comment (saghul) [inline debian/jitsi-meet-web-config.postinst]: @damencho what are your thoughts on this? On one had it makes the setup multi-domain friendly, OTOH, more setup would be required for that (AFAICT) so I'd rather have a sed there...

**Comment by damencho** [comment]
I just found a case that this doesn't work. It is the webpack-dev-server which developers are using a lot. 
I was also wondering about the react-mobile part, @saghul, is the mobile currently downloading config.js, what will happen in there?

**Comment by saghul** [comment]
@damencho it just won't work, you're right.  This PR had good intentions, but I'm afraid it makes many cases more complicated than they need to be.

@pierreozoux I'm going to close this based on the discussion above. For Docker style deployments, you can use sed like we do in the Debian package to replace jisti.example.com with window.location.hostname, for example.

**Pull request closed without merging**